### ISSUE TITLE
Environment variables from file rather than VSCode settings

### ISSUE THREAD

**Issue by chaeron** [Huachao/vscode-restclient#673]
Having environment variables defined in VSCode setting works, but is awkward when you are working on a multi-person project since it's harder to share the settings with your colleagues, save them in Git, version them, etc.

Would be really nice if there was a way to load different environments from a separate file rather than settings.

**Comment by madiken21** [comment]
+1

**Comment by Marvin-Brouwer** [comment]
+1 
Maybe just recurse up from http files finding a environment.json that overrides it's parent?

**Comment by mchelen-gov** [comment]
Have you looked at the dotenv variables like `{{$dotenv USERNAME}}`?

**Comment by Marvin-Brouwer** [comment]
@mchelen-gov That only works from the same folder as where the files are in.
Also, it doesn't allow you to change environments using the editor UI. You'll end up having "environment-a.env" and renaming them when you need them.

**Comment by iainwlg** [comment]
Yes please. This would be great timesaver when sharing a .http collection.
